tsdb/engine/tsm1/cache.go

```diff
@@ -252,29 +253,24 @@ func (c *Cache) Snapshot() (*Cache, error) {
 	// If no snapshot exists, create a new one, otherwise update the existing snapshot
 	if c.snapshot == nil {
 		c.snapshot = &Cache{
-			store: make(map[string]*entry),
+			store: NewCacheStoreMap(),
 		}
 	}
 
 	// Append the current cache values to the snapshot
-	for k, e := range c.store {
-		e.mu.RLock()
-		if _, ok := c.snapshot.store[k]; ok {
-			c.snapshot.store[k].add(e.values)
-		} else {
-			c.snapshot.store[k] = e
-		}
-		c.snapshotSize += uint64(Values(e.values).Size())
-		if e.needSort {
-			c.snapshot.store[k].needSort = true
-		}
-		e.mu.RUnlock()
+
+	// this is the highest-value thing we can do on the write path, so
+	// dedicate many resources to it:
+	nWorkers := runtime.GOMAXPROCS(0) / 2
```

Comment:
What benchmarks drove the choice of this value, seems quite arbitrary, may be better to have a configuration file option for it.